tensorly/tenalg/proximal.py

```diff
@@ -248,3 +248,207 @@ def hals_nnls(UtM, UtU, V=None, n_iter_max=500, tol=10e-8,
                 break
 
     return V, rec_error, iteration, complexity_ratio
+    
+
+def fista(AtB, pseudo_inverse, x=None, n_iter_max=100, non_negative=True, gradient_step=None,
+          sparsity_coefficient=None):
+    """
+    Fast Iterative Shrinkage Thresholding Algorithm (FISTA)
+
+    Computes and approximate solution for Ax=b linear system.
+
+    Parameters
+    ----------
+    AtB: ndarray
+       Pre-computed product of the transposed of A and B.
+    pseudo_inverse: ndarray
+       Pre-computed product of the transposed of A and A.
+    x: initialized array
+       Default: None
+    n_iter_max : int
+        Maximum number of iteration
+        Default: 100
+    non_negative : bool, default is False
+                   if True, result will be non-negative
+    gradient_step : float
+    sparsity_coefficient : float or None
+
+    Returns
+    -------
+    x : Updated ndarray
+
+    Reference
+    ----------
+    [1] : Beck, A., & Teboulle, M. (2009). A fast iterative
+          shrinkage-thresholding algorithm for linear inverse problems.
+          SIAM journal on imaging sciences, 2(1), 183-202.
+    """
+
+    if sparsity_coefficient[-1] is None:
+        sparse = 0
+    else:
+        sparse = sparsity_coefficient[-1]
+
+    if gradient_step is None:
+        gradient_step = 0.001
+
+    if x is None:
+        x = tl.zeros([tl.shape(pseudo_inverse)[0], tl.shape(AtB)[1]])
+
+    # Parameters
+    momentum_old = tl.tensor(1.0)
+    norm_0 = 0.0
+    x_upd = tl.copy(x)
+
+    for iteration in range(n_iter_max):
+        gradient = - AtB + tl.tenalg.multi_mode_dot(x_upd, pseudo_inverse, transpose=False) + sparse
+
+        if non_negative is True:
+            delta_x = tl.where(gradient_step * gradient < x, gradient_step * gradient, x_upd)
+        else:
+            delta_x = gradient_step * gradient
+
+        xnew = x_upd - delta_x
```

Comment:
```suggestion
        x_new = x_update - gradient_step*x_gradient
```

nitpick: x_new to be consistent with rest?
I would also remove the gradient step from above and make it here for clarity.